src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -69,6 +84,144 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         {
             _lastStartingDirectoryPath = StartingDirectory();
         }
+
+        // generate the font list, if we don't have one
+        if (!_FontList || !_MonospaceFontList)
+        {
+            UpdateFontList();
+        }
+    }
+
+    // Method Description:
+    // - Updates the lists of fonts and sorts them alphabetically
+    void ProfileViewModel::UpdateFontList() noexcept
+    try
+    {
+        // initialize font list
+        std::vector<hstring> fontList;
+        std::vector<hstring> monospaceFontList;
+
+        // get a DWriteFactory
+        com_ptr<IDWriteFactory> factory;
+        THROW_IF_FAILED(DWriteCreateFactory(
+            DWRITE_FACTORY_TYPE_SHARED,
+            __uuidof(IDWriteFactory),
+            reinterpret_cast<::IUnknown**>(factory.put())));
+
+        // get the font collection; subscribe to updates
+        com_ptr<::IDWriteFontCollection> fontCollection;
+        THROW_IF_FAILED(factory->GetSystemFontCollection(fontCollection.put(), TRUE));
+
+        for (UINT32 i = 0; i < fontCollection->GetFontFamilyCount(); ++i)
+        {
+            try
+            {
+                // get the font family
+                com_ptr<::IDWriteFontFamily> fontFamily;
+                THROW_IF_FAILED(fontCollection->GetFontFamily(i, fontFamily.put()));
+
+                // get the font's localized names
+                com_ptr<::IDWriteLocalizedStrings> localizedFamilyNames;
+                THROW_IF_FAILED(fontFamily->GetFamilyNames(localizedFamilyNames.put()));
+
+                // use our current locale to find the correct name
+                BOOL exists{ FALSE };
+                UINT32 index;
+                HRESULT hr;
+                wchar_t localeName[LOCALE_NAME_MAX_LENGTH];
+                if (GetUserDefaultLocaleName(localeName, LOCALE_NAME_MAX_LENGTH))
+                {
+                    hr = localizedFamilyNames->FindLocaleName(localeName, &index, &exists);
+                }
+                if (SUCCEEDED(hr) && !exists)
+                {
+                    hr = localizedFamilyNames->FindLocaleName(L"en-us", &index, &exists);
+                }
+                if (!exists)
+                {
+                    // failed to find the correct locale, using the first one
+                    index = 0;
+                }
+
+                // get the localized name
+                UINT32 nameLength;
+                THROW_IF_FAILED(localizedFamilyNames->GetStringLength(index, &nameLength));
+                wchar_t* localizedName = new (std::nothrow) wchar_t[nameLength + 1];
```

Comment:
just reattaching this comment here because moving the source around made it disappear.

![image](https://user-images.githubusercontent.com/189190/109343672-8fb29a00-7822-11eb-97cf-f3c622db1de9.png)